smartsim/_core/launcher_/slurm/slurm_launcher.py

```diff
@@ -0,0 +1,334 @@
+            should be aware of. Primarily used for testing.
+        """
+        self._launched: t.Final = dict(launched) if launched is not None else {}
+
+    @classmethod
+    def create(cls, _: Experiment) -> SlurmLauncher:
+        """Create a new launcher instance from an experiment instance.
+
+        :param _: <Unused> An experiment instance.
+        :returns: A new launcher instance.
+        """
+        return cls()
+
+    def start(self, launchable: _SlurmCommandType, interval: int = 2) -> LaunchedJobID:
+        """Have the slurm launcher start and track the progress of a new
+        subprocess.
+
+        :param launchable: The template of a slurm subprocess to start.
+        :param interval: The amount of time in seconds to wait between `sacct`
+            calls to get the step id.
+        :returns: An id to reference the process for status tracking.
+        """
+        launchable.start()
+        trials = CONFIG.wlm_trials
+        step_id = None
+        while step_id is None and trials > 0:
+            # >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+            # TODO: Really don't like the implied sacct format and required
+            #       order to call these fns in order to get the step id
+            # -------------------------------------------------------------------------
+            out, _ = commands.sacct(
+                ["--noheader", "-p", "--format=jobname,jobid"], raise_on_err=True
+            )
+            step_id = parser.parse_step_id_from_sacct(out, launchable.name)
+            # <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+            trials -= 1
+            if step_id is None:
+                time.sleep(interval)
+        if step_id is None:
+            raise errors.LauncherError("Could not find id of launched job step")
+        id_ = launcher.create_job_id()
+        self._launched[id_] = _LaunchedJobInfo(step_id, launchable.name)
+        return id_
+
+    def _get_slurm_info_from_job_id(self, id_: LaunchedJobID, /) -> _LaunchedJobInfo:
+        """Find the info for a slurm subprocess given a launched job id issued
+        to a user.
+
+        :param id_: The job id issued to the user.
+        :raises errors.LauncherJobNotFound: The id is not recognized.
+        :returns: Info about the launched slurm job.
+        """
+        if (info := self._launched.get(id_)) is None:
+            msg = f"Launcher `{self}` has not launched a job with id `{id_}`"
+            raise errors.LauncherJobNotFound(msg)
+        return info
+
+    def get_status(self, *ids: LaunchedJobID) -> t.Mapping[LaunchedJobID, JobStatus]:
+        """Take a collection of job ids and return the status of the
+        corresponding slrum processes started by the slurm launcher.
+
+        :param ids: A collection of ids of the launched jobs to get the
+            statuses of.
+        :returns: A mapping of ids for jobs to stop to their reported status.
```

Comment:
I have no idea what "...to stop to their reported status" means.. copypasta error from a method that cancels jobs, maybe?

Reply:
copypasta error 🤦. Replaced with wording matching the launcher base class